pkg/cache/unavailableofferings.go

```diff
@@ -69,6 +69,15 @@ func (u *UnavailableOfferings) MarkUnavailable(ctx context.Context, unavailableR
 	u.MarkUnavailableWithTTL(ctx, unavailableReason, instanceType, zone, capacityType, UnavailableOfferingsTTL)
 }
 
+func (u *UnavailableOfferings) Put(key string, value struct{}, ttl time.Duration) {
+	u.cache.Set(key, value, ttl)
+}
+
+func (u *UnavailableOfferings) Has(key string) bool {
+	_, found := u.cache.Get(key)
+	return found
+}
```

Comment:
This seems to expose too much of the implementaional detail of UnavailableOfferings - and no reason to pass value? How about adding a method like MarkSpotUnavailableWithTTL, instead of opening the cache for direct manipulation? It could also  maybe use synthetic key("","",spot), instead of the specific error string?

Reply:
synthetic key makes a lot of sense, will switch to that

Reply:
The new changes go in the right direction I think, but I would suggest taking the next step and 

* Implement MarkSpotUnavailableWithTTL method on UnavailableOfferings (that would use the synthetic key internally) - this would remove from the consume the need to know anything about implementation; and 
* Add the check for this key into the already existing IsUnavailable (would check for this first, if asked about spot), which the consumer would call as before, without special casing / knowing anything about extra check for spot behind the scenes.

Reply:
gotcha that makes sense